core/src/main/java/org/elasticsearch/index/seqno/GlobalCheckpointService.java

```diff
@@ -152,9 +152,6 @@ synchronized void updateCheckpointOnReplica(long globalCheckpoint) {
         if (this.globalCheckpoint <= globalCheckpoint) {
```

Comment:
can you add a comment about when the current global checkpoint can be higher? here is what I wrote in #10708 

> Note that the global checkpoint is a local knowledge of that is update under the mandate of the primary. It may be that the primary information is lagging compared to a replica. This can happen when a replica is promoted to a primary (but still has stale info).